climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1387 @@
+            csv_firms : pd.DataFrame or str
+                 path to csv file of FIRMS data or FIRMS data as pd.Dataframe
+
+        Returns
+        -------
+            res_data/ONE_LAT_KM : float
+                resolution in degrees
+
+        """
+        # Resolution in km of the centroids depends on the data origin.
+        if 'instrument' in firms.columns:
+            if firms['instrument'].any() == 'MODIS':
+                res_data = 1.0
+            else:
+                res_data = 0.375 # For VIIRS data
+        else:
+            res_data = RES_DATA # For undefined data origin, defined by user
+        return res_data/ONE_LAT_KM
+
+    @staticmethod
+    def _firms_centroids_creation(firms, res_data, centr_res_factor):
+        """ Get centroids from the firms dataset and refactor them.
+
+        Parameters
+        ----------
+            firms : pd.DataFrame
+                 FIRMS data
+            res_data : float
+                FIRMS instrument resolution in degrees
+            centr_res_factor : float
+                the factor applied to voluntarly decrease/increase the
+                centroids resolution
+
+        Returns
+        -------
+            centroids : Centroids
+
+        """
+        centroids = Centroids()
+        centroids.set_raster_from_pnt_bounds((firms['longitude'].min(), \
+            firms['latitude'].min(), firms['longitude'].max(), \
+            firms['latitude'].max()), res=res_data/centr_res_factor)
+        centroids.set_meta_to_lat_lon()
+        centroids.set_area_approx()
+        centroids.set_on_land()
+        centroids.empty_geometry_points()
+
+        return centroids
+
+    @staticmethod
+    def _centroids_resolution(centroids):
+        """ Return resolution of the centroids in their units
+
+        Parameters
+        ----------
+            centroids (Centroids): centroids instance
+
+        Returns
+        -------
+            res_centr : float
+                grid resolution of centroids
+        """
+        if centroids.meta:
+            res_centr = abs(centroids.meta['transform'][4]), \
+                centroids.meta['transform'][0]
+        else:
+            res_centr = u_coord.get_resolution(centroids.lat, centroids.lon)
+        if abs(abs(res_centr[0]) - abs(res_centr[1])) > 1.0e-6:
+            LOGGER.warning('Centroids do not represent regular pixels %s.', str(res_centr))
+            return (res_centr[0] + res_centr[1])/2
+        return res_centr[0]
+
+    def _firms_cons_days(self, firms):
+        """ Compute clusters of consecutive days (temporal clusters).
+        
+            An interruption of self.days_thresh_firms is necessary to be set
+            in two different temporal clusters.
+
+        Parameters
+        ----------
+            firms : pd.DataFrame
+                 FIRMS data
+
+        Returns
+        -------
+            firms : pd.DataFrame
+                FIRMS data including info on temporal cluster per point
```

Comment:
This is a very dangerous method! It modifies inplace the input dataframe. I very strongly advise against such a practice. 
At the very very minimum, and I am not sure whether this is enough, this must be very very clearly stated in the docstring. Better would be to not modify the dataframe in place, but to return a copy.

Reply:
I've adjusted the docstring 7cd85afa